siliconcompiler/core.py

```diff
@@ -3855,11 +3871,18 @@ def _setup_tool(self, tool, task, step, index):
         self.set('arg','step', step)
         self.set('arg','index', index)
 
-        func = self.find_function(tool, 'setup', 'tools')
-        if func is None:
-            self.logger.error(f'setup() not found for tool {tool}')
+        # Run task setup.
+        try:
+            task = self.get('flowgraph', self.get('option', 'flow'), step, index, 'task')
```

Comment:
use `get_task`

Reply:
Any thoughts on the TODO questions for `get_task`? (Should we also add `get_tool`, add `_step_` to the name(s), and do we want these small helpers in the Chip object?)

Reply:
I don't have any strong feelings about it, but maybe @aolofsson want to? or we can collect this for a future API review

Reply:
Okay - I'll rename it to `_get_task`, and we can hold off on `_get_tool`.